### PR TITLE
✨Add ontology prefix validation

### PR DESCRIPTION
**Pull request by parimalak**
validation for ontology columns

### CODE REVIEW

File: dataservice/api/common/validation.py [hunk 1]
```diff
@@ -47,3 +49,53 @@ def enum_validation_generator(_enum):
...
+        """Perform validation on ``value``. Raise a :exc:`ValidationError`
+        if validation
+        does not succeed.
+        """
+        errors = []
+        kwargs = {}
+        for validator in self.validators:
+            try:
+                try:
+                    r = validator(value, field=self)
```
Comment (parimalak) [inline dataservice/api/common/validation.py:69]: we can remove this `FieldValidator` class once this PR https://github.com/marshmallow-code/marshmallow/issues/289 for marshmallow is merged. in this we are extending `_validate` function to allow field name to the custom validator function

File: dataservice/api/common/validation.py [hunk 2]
```diff
@@ -47,3 +49,53 @@ def enum_validation_generator(_enum):
...
+    onto_dict = {
+        'mondo': 'MONDO:',
+        'uberon': 'UBERON:',
+        'icd':	'ICD10:',
+        'hpo':	'HP:',
+        'snomed': 'SNOMEDCT:',
+        'ncit': 'NCIT:'
+    }
+    if attribute is not None:
+        if attribute.startswith(onto_dict[prefix]) is False:
```
Comment (dankolbman) [inline dataservice/api/common/validation.py:99]: Probably best to check that prefix is in the onto_dict?

File: dataservice/api/diagnosis/schemas.py [hunk 3]
```diff
@@ -20,6 +22,15 @@ class DiagnosisSchema(BaseSchema):
...
                                        DIAGNOSIS_CATEGORY_ENUM))
+    mondo_id_diagnosis = FieldValidator(attribute='mondo_id_diagnosis',
+                                        validate=validate_ontology_id_prefix)
+    icd_id_diagnosis = FieldValidator(attribute='icd_id_diagnosis',
+                                      validate=validate_ontology_id_prefix)
+    uberon_id_tumor_location = FieldValidator(
+        attribute='uberon_id_tumor_location',
+        validate=validate_ontology_id_prefix)
+    ncit_id_diagnosis = FieldValidator(attribute='ncit_id_diagnosis',
+                                       validate=validate_ontology_id_prefix)
```
Comment (dankolbman) [inline dataservice/api/diagnosis/schemas.py:37]: maybe `validate_ontology_id_prefix` could be a generator that takes a prefix and returns a function that validates that field using the prefix? That way lookup dict won't be needed and you won't have to pass attributes to it.

Comment (znatty22) [reply dataservice/api/diagnosis/schemas.py:37]: 👍 I agree thats what I would suggest as well ^

**Pull request closed without merging**